python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -212,7 +212,7 @@ def __call__(self, fn: Callable) -> AssetsDefinition:
                 required_resource_keys.add(key)
 
             op = _Op(
-                name="__".join(out_asset_key.path),
+                name="__".join(out_asset_key.path).replace("-", "_"),
```

Comment:
might we want to do this replacement at the time we construct the asset key?

Reply:
I think that might be a bit too aggressive, as there is a difference between ["snowflake", "test-schema', "foo_table"] and ["snowflake", "test_schema", "foo_table"]